crates/builder/src/tasks/submit.rs

```diff
@@ -12,18 +12,28 @@ use alloy::signers::Signer;
 use alloy::sol_types::SolCall;
 use alloy::transports::TransportError;
 use alloy_primitives::{FixedBytes, U256};
+use alloy_sol_types::SolError;
 use eyre::{bail, eyre};
 use oauth2::{
     basic::BasicClient, basic::BasicTokenType, reqwest::http_client, AuthUrl, ClientId,
     ClientSecret, EmptyExtraTokenFields, StandardTokenResponse, TokenResponse, TokenUrl,
 };
 use tokio::{sync::mpsc, task::JoinHandle};
 use tracing::{debug, error, instrument, trace};
-use zenith_types::{SignRequest, SignResponse, Zenith};
+use zenith_types::{
+    SignRequest, SignResponse,
+    Zenith::{self, IncorrectHostBlock},
+};
 
 /// OAuth Audience Claim Name, required param by IdP for client credential grant
 const OAUTH_AUDIENCE_CLAIM: &str = "audience";
 
+pub enum ControlFlow {
+    Retry,
+    Skip,
```

Comment:
skip is never constructed. we need to convert things that would previously `?` or `return Err` to return `Ok(Skip)`